### PR TITLE
tooltip shows all coupled files

### PR DESCRIPTION
**Pull request by bieron**
Hello. Great work!

nodes in cluster colored distinctly now
- disabled radius growth on hover because it clashed with color change
  and I couldn't get it to work properly
- removed style for .node:hover

You will see that my algorithm for clustering the graph is pretty dumb and costly, but it works. I also tried to keep to your coding style. I am not a professional JavaScripter, so if you don't feel like merging my feature in its current state, I won't hold it against you :)

### CODE REVIEW

**Comment by pnavarrc** [comment]
Hi, thanks for taking the time to create the pull request! As you probably noticed, this project is not very active, but the problem is certainly interesting, your contribution is very appreciated.

I will review and merge the pull request shortly, I will also take the opportunity to update the readme and include some screenshorts including your changes. Thanks again for contributing!

File: app/scripts/main.js [hunk 1]
```diff
@@ -24,33 +22,94 @@ function computeNetwork(data, correlationThreshold) {
 
 	var nodeMap = {};
 
+	// Variables for clustering and coloring
+	var clusters = [],
+    // TODO consistent colorscheme
+        colors = [
```
Comment (pnavarrc) [inline app/scripts/main.js]: Maybe you can use https://github.com/mbostock/d3/wiki/Ordinal-Scales#category20 ?

File: app/scripts/main.js [hunk 2]
```diff
@@ -24,33 +22,94 @@ function computeNetwork(data, correlationThreshold) {
...
-	
+			target = nodeMap[row.fileB];
+
+		var n1, n2, c1, c2;
+
 		if (row.corr > correlationThreshold) {
-			network.links.push({source: source, target: target});			
+			links.push({source: source, target: target});
+
+            n1 = nodes[source];
```
Comment (pnavarrc) [inline app/scripts/main.js]: Maybe keep the indentation to 2 spaces.

File: app/scripts/main.js [hunk 3]
```diff
@@ -24,33 +22,94 @@ function computeNetwork(data, correlationThreshold) {
...
+                clusters[ clusters.indexOf(c1) ] = newCluster;
+                clusters[ clusters.indexOf(c2) ] = undefined;
+            }
 		}
-
 	});
 
-	return network;
+    _.each(clusters, function(c,i) {
+        if (typeof c === 'undefined') {
```
Comment (pnavarrc) [inline app/scripts/main.js]: It might be more easy to read (and write) if using `_.isUndefined` instead

File: app/scripts/main.js [hunk 4]
```diff
@@ -24,33 +22,94 @@ function computeNetwork(data, correlationThreshold) {
...
 		if (row.corr > correlationThreshold) {
-			network.links.push({source: source, target: target});			
+			links.push({source: source, target: target});
+
+            n1 = nodes[source];
+            n2 = nodes[target];
+            c1 = n1.cluster;
+            c2 = n2.cluster;
+
+            if (typeof c1 === 'undefined' && typeof c2 === 'undefined') {
```
Comment (pnavarrc) [inline app/scripts/main.js]: Maybe using `_.isUndefined` might be better here (and below)

File: app/scripts/main.js [hunk 5]
```diff
@@ -24,33 +22,94 @@ function computeNetwork(data, correlationThreshold) {
...
+        if (typeof c === 'undefined') {
+            return;
+        }
+
+		var group = {
+			tooltip: [],
+			name: 'cluster'+ i
+		};
+
+        _.each(c, function(n, i) {
```
Comment (pnavarrc) [inline app/scripts/main.js]: Maybe fix the indentation a bit (2 spaces for each indentation level)

File: less/main.less [hunk 6]
```diff
@@ -20,16 +20,10 @@ svg.network-chart {
...
-	}
 }
 
 // Links
 
 .link {
 	stroke: @grey-medium;
-}
\ No newline at end of file
+}
```
Comment (pnavarrc) [inline less/main.less:29]: :+1:

**Comment by pnavarrc** [comment]
I added some comments, the code looks good overall, just some minor style suggestions. If you don’t have time to change it just let me know and I will do it, thanks for your contribution again!

**Comment by bieron** [comment]
Here you go :)

**Pull request merged**